Bux fix jump animation and rocks expoiding

Ian-kowalski opened the pull request.

Ian-kowalski committed:
Synchronize jump physics with jump animation

Added helper methods to AnimationController to detect jump state and normalized time. Updated MovementScript to trigger jump animation and apply jump physics after the animation starts, ensuring better synchronization between animation and movement.

Ian-kowalski committed:
Add dust poof animation to rock interaction

Enhanced the rock interactable to play a dust poof animation when interacted with, using an Animator and associated SpriteRenderers. The script now manages animator state, triggers, and disables the rock sprite after the animation completes. Also added debug logging and improved state saving/loading to support the new animation logic.

Ian-kowalski committed:
Refactor rock initialization and state loading

Moved initialization logic from Start to Awake and added EnsureInitialized to guarantee references are set before state loading. Improved LoadState to check uniqueId before applying state, reset poof visuals, and prevent poof animation replay on load. These changes improve reliability when loading and saving rock state.

Ian-kowalski committed:
Refactor jump logic to sync animation and physics

Jump physics are now applied in sync with the jump animation, using a coroutine to wait for the animation state and handle timeouts. Added a configurable jump state name and an option to apply jump from an animation event, improving reliability and flexibility of jump behavior.

Ian-kowalski committed:
Refactor jump logic for immediate physics application

Jump physics are now applied synchronously with the animation trigger in jumpAction(), improving responsiveness and test reliability. The ground check null handling is improved to avoid disabling the component, and test helpers ensure jump velocity is set immediately for accurate test results.

Ian-kowalski committed:
Refactor test jump helpers in MovementScript

Split Test_Jump into two methods: one that simulates a jump button press respecting coyote time, and another (Test_Jump_ForceCoyote) that forces coyote time for tests needing immediate jump regardless of grounding. This improves test reliability and flexibility.

Ian-kowalski committed:
Update MovementScript.cs

Ian-kowalski commented on Assets/Scripts/Player/AnimationController.cs:
```diff
@@ -12,6 +12,9 @@ public class AnimationController : MonoBehaviour
         private static readonly string IS_IDLE = "IsIdle";
 
+        [Header("State Names")] [SerializeField]
+        private string jumpStateName = "Jump";
```
@copilot open a new pull request to apply changes based on [this feedback](https://github.com/Ian-kowalski/Consume-Clearly/pull/48#discussion_r2584294517)

Copilot committed:
Initial plan

Copilot committed:
Make jumpStateName readonly by converting to static readonly constant

Co-authored-by: Ian-kowalski <85852048+Ian-kowalski@users.noreply.github.com>

Ian-kowalski committed:
Remove Test scene and metadata from project

Deleted the Test.unity scene and its associated Test.unity.meta file from the Assets/Scenes directory. This change likely reflects cleanup of unused or obsolete content.

kentsoop reviewed (commented):
I dont know if this can be solved here but we have a delay between the jump and the start of the animation

NotJezus reviewed (approved):
nice,  but look at ai and the test please?

Ian-kowalski committed:
Merge pull request #49 from Ian-kowalski/copilot/sub-pr-48

Make jumpStateName field readonly

Ian-kowalski committed:
Refactor rock class and optimize poof animation logic

Renamed 'rock' to 'Rock' for consistency and refactored poof animation trigger handling by caching trigger existence and hash for performance. Removed redundant HasAnimatorTrigger method, improved state loading logic, and cleaned up debug logging. Also removed unnecessary jump animation debug logs from MovementScript.

NotJezus commented on Assets/Scripts/LevelObjects/Interactable/rock.cs:
```diff
@@ -37,12 +165,36 @@ public override void LoadState(InteractableObjectState state)
             if (state == null) return;
 
+            // Only load state intended for this object
+            if (string.IsNullOrEmpty(state.uniqueId) || state.uniqueId != GetUniqueId())
```
can be written shorter

NotJezus reviewed (approved):
nice

Ian-kowalski committed:
Merge branch 'main' into bux-fix-Jump-animation-and-rocks-expoiding

Ian-kowalski committed:
Create Test.unity.meta
